start.sh

```diff
@@ -3,27 +3,101 @@ set -ex
 
 GROUP=staff
 USER_UID=1001
+export HOME="/home/$USER"
 
-if [ "$(id -u "$USER" 2>/dev/null)" != $USER_UID ]; then
-    useradd -g $GROUP -u $USER_UID -d /home/$USER $USER
-fi
+function init_user() {
+  if [ "$(id -u "$USER" 2>/dev/null)" != $USER_UID ]; then
+      useradd -g $GROUP -u $USER_UID -d "/home/$USER" "$USER" || true
+  fi
 
-# this is only for running locally - in-cluster, home directories are
-# provided via NAS
-echo "${USER}:${USER}" | chpasswd
-if [ ! -d /home/$USER ]; then
-    mkdir -p /home/$USER
-fi
-chown "${USER}:${GROUP}" /home/$USER
-export HOME=/home/$USER
+  # this is only for running locally - in-cluster, home directories are
+  # provided via NAS
+  echo "${USER}:${USER}" | chpasswd
+  if [ ! -d "/home/$USER" ]; then
+      mkdir -p "/home/$USER"
+      chown "${USER}:${GROUP}" -R "/home/$USER/"
+  fi
+}
+
+function init_conda() {
+  ## if user has ~/.bashrc make sure conda is added to that
+
+  CONDA_SNIPPET='[[ -f /opt/conda/etc/profile.d/conda.sh ]] && . /opt/conda/etc/profile.d/conda.sh'
+  CONDA_ENV_ACTIVATE='[[ -v RSTUDIO ]] && conda activate rstudio'
+
+  grep -q -x -F "$CONDA_SNIPPET" "$HOME/.bashrc" || echo "$CONDA_SNIPPET" >> "$HOME/.bashrc"
+  grep -q -x -F "$CONDA_ENV_ACTIVATE" "$HOME/.bashrc" || echo "$CONDA_ENV_ACTIVATE" >> "$HOME/.bashrc"
+
+  # shellcheck disable=SC1091
+  . /opt/conda/etc/profile.d/conda.sh
+
+    conda create --use-index-cache --clone root -n rstudio --copy \
```

Comment:
We recently saw errors during creation when the path partially existed.  i.e if `~/.conda/whatever` existed `conda create` would fail.  Maybe `stat` or `test -d` for the enth directory you're expecting 🤷‍♂️

Reply:
changing to use `conda list envs` because if it's a random directory with a failed conda install then we just want to overwrite